luigi/scheduler.py

```diff
@@ -862,9 +885,8 @@ def _upstream_status(self, task_id, upstream_status_table):
                     elif upstream_status_table[dep_id] == '' and dep.deps:
                         # This is the postorder update step when we set the
                         # status based on the previously calculated child elements
-                        status = max((upstream_status_table.get(a_task_id, '')
-                                      for a_task_id in dep.deps),
-                                     key=UPSTREAM_SEVERITY_KEY)
+                        upstream_severities = list(upstream_status_table.get(a_task_id) for a_task_id in dep.deps if a_task_id in upstream_status_table) or ['']
+                        status = min(upstream_severities, key=UPSTREAM_SEVERITY_KEY)
```

Comment:
Here is the change of luigi default behavior to detect upstream severity. Parent task marked as `UPSTREAM_DISABLED` when any of its upstream task is `DISABLED` and the worker was shutdown because of it. This behavior is changed to wait for all upstream tasks to be `DISABLED` or `FAILED` to mark parent task as `UPSTREAM_DISABLED` or `UPSTREAM_FAILED`.